semanticcpg/src/main/scala/io/shiftleft/semanticcpg/accesspath/TrackingPointToAccessPath.scala

```diff
@@ -9,101 +9,180 @@ import org.slf4j.LoggerFactory
 
 import scala.jdk.CollectionConverters._
 
-object TrackingPointToAccessPath {
-
-  def apply(node: nodes.TrackingPoint, exclusions: List[Elements] = List()): AccessPath = {
-    new AccessPath(TrackingPointToElements(node), exclusions)
-  }
-
+trait FakeTrackingPoint {
```

Comment:
So, can you explain again why we need this fake tracking point? The reason I ask is that anyone reading this will ask themselves in which way this tracking point is fake. The reason I didn't originally port over these fake tracking points is because it seemed conceptually unclean to me.

Reply:
They _are_ conceptually unclean -- the underlying architecture wants all subclasses of `node` to be generated and known to odb, i.e. these should really be `sealed`.

However, we have several APIs that expect `node` that turned out to sometimes require extra data, and therefore we have a bunch of fake nodes all over the place (most prevalent in the closed source data flow tracker) -- that is, classes that claim to implement out node interfaces, but in reality only implement a tiny subset, and will blow up if they get serialized (e.g. because they are part of a Flow).

For this reason, codescience currently needs its own implementation (API entry-point) for all these methods, in order to `match` over the fake node classes. An effect of that is that it is hard to share code between the open and closed tracker (in the open source world, you use these methods for accessing base+path, in the closed source world you use these completely different methods).

That is bonkers, so it's better to define one extra interface for fake tracking points and use that. Yes, it's unclean, but this way we at least contain the uncleanliness.

I'll add source code comments to that effect.